pages/article/[pid].js

```diff
@@ -0,0 +1,48 @@
+import { useState, useEffect } from 'react';
+import { useRouter } from 'next/router';
+import { collection, onSnapshot, query, where } from 'firebase/firestore';
+import Link from 'next/link';
+
+import { db } from '../../firebase/initFirebase';
+
+const Article = () => {
+  const router = useRouter();
+  const { pid } = router.query;
+
+  const [article, setArticle] = useState([]);
+
+  useEffect(() => {
+    if (router.isReady) {
+      const articlesCollectionRef = collection(db, 'articles');
+      const q = query(articlesCollectionRef, where('link', '==', pid));
+
+      onSnapshot(q, (snapshot) => {
```

Comment:
What is the difference between mine `getDocs` and your `onSnapshot` https://github.com/GLObus303/showcase-project/blob/master/src/api/rooms.ts#L25 ?

Reply:
getDocs - get data once - https://firebase.google.com/docs/firestore/query-data/get-data#web-version-9

onSnapshot - realtime updates listener - https://firebase.google.com/docs/firestore/query-data/listen#web-version-9

Reply:
I used the getDocs in api. I couldn't make it work with onSnapshot, don't know why.

Reply:
I dont think we need real-time updates :D. What did not work with the `getDocs` ?

Reply:
Everything was ok, I only decided to use the second method :D, but I re-created it to getDocs, as you will see..